test/fetch-ec-data_spec.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+eval "$(shellspec -)"
+
+# Avoid running real tkn command in fetch.sh
+tkn() {
+  echo "dummy-tkn-response"
+}
+
+Include ./appstudio-utils/util-scripts/lib/fetch.sh
+
+Describe 'json-data-file'
+  local root_dir=$( git rev-parse --show-toplevel )
+  local data_dir="/tmp/ecwork/data"
+
+  It 'produces expected json file paths'
+    When call json-data-file some dir 123 foo 456
+    The output should eq "$data_dir/some/dir/123/foo/456/data.json"
+  End
+End
+
+Describe 'rekor-log-entry'
+  rekor-cli() {
+    echo "rekor-cli $*"
+  }
+
+  It 'calls rekor-cli as expected'
+    When call rekor-log-entry 1234 rekor.example.com
+    The output should eq "rekor-cli get --log-index 1234 --rekor_server https://rekor.example.com --format json"
+  End
+End
+
+# We're not testing the data itself, but I think it's okay
+#
+Describe 'rekor-log-entry-save'
+  json-data-file() {
+    echo "/dev/null"
+  }
+
+  # Mock some fake rekor data
+  rekor-log-entry() {
+    if [[ $1 == 1235 ]]; then
+      # With an attestation
+      echo '{"foo":"bar","Attestation":"'$( echo '{"hey":"now"}' | base64 )'"}'
+    else
+      # Without an attestation
+      echo '{"foo":"bar","Attestation":""}'
+    fi
+  }
+
+  It 'saves rekor data without an attestation'
+    When call rekor-log-entry-save 1234 rekor.example.com
+    The output should eq "Saving log index 1234 from rekor.example.com"
+  End
+
+  It 'saves rekor data with an attestation'
+    When call rekor-log-entry-save 1235 rekor.example.com
+    The line 1 should eq "Saving log index 1235 from rekor.example.com"
+    The line 2 should eq "Saving attestation extracted from rekor data"
+  End
+
+
+  # Put data in a file so we can look at it
+  local tmp_file=$(mktemp)
```

Comment:
Just to note `shellspec` can be run with `--tmpdir /somedir` and with `--keep-tmpdir`, but for that to work we need:
```suggestion
  local tmp_file=$(mktemp --tmpdir)
```